### ISSUE TITLE
try to make it exciting to run - brainstorming

### ISSUE THREAD

**Issue by ornicar** [niklasf/fishnet#20]
I like things that grow. I like being rewarded constantly.

Can we count and display stuff while fishnet runs? Like, sum of moves analysed, nodes searched, evaluation depth. Give me something to look at and brag about.

Also lichess could collect these stats on the server and send them to the client on start, for persistence.

**Comment by clarkerubber** [comment]
1. Proportion of total analyses provided.
2. "Without your support, analyses would take on average X seconds longer (average analysis time is Y seconds)." (not sure if possible, likely going to need to whip the calculator out)
3. Load vs Time
4. Accumulative analyses produced / Time
5. Analyses produced vs Time
6. Total number of unique positions analysed

**Comment by mekhami** [comment]
Let people attach their account to the fishnet servers their running, then give them a badge on their profile that shows how many analyses they've processed. That's motivation enough for me!

**Comment by verdug0** [comment]
Let spray bitcoins on each analisys!

Ok, What if starts with lichess showing us more data about the actual fishnet, perhaps something about nodes, contribution of each, in analisys, moves, uptime, etc.

What if lichess just show to the community the info, finding new enthusiasts to contribute with new and bitcoined(ok, stop it!) ideas.

FishNet can grow up(ex:in node count) if it can to show to the community a challenge that We can resolve just together.

Let start finding a good chess challenge too!

**Comment by ddugovic** [comment]
Maybe allow the players to see which game analyses they contributed to...

**Comment by apetresc** [comment]
@ddugovic: The standard output of fishnet already spits out links to every game you analyze. Unless you mean, like, a list on the website somewhere, but that would be _thousands_ of games for most people, probably not a good idea.

I *love* the idea of collecting aggregate stats (like total nodes/hours/etc) and displaying them both in fishnet and on the user profile, though.

**Comment by ddugovic** [comment]
@apetresc Yes, I'm suggesting a list, although from this thread it sounds like somehow people might be satisfied with aggregate numbers.

**Comment by apetresc** [comment]
@ddugovic If you really want a list of the games it's analysed, just run fishnet with the command:
```
$ python -m fishnet run | tee -a out.log
```

And then just go:
```
$ cat out.log| grep -oP "(http://.*) took" | grep -o "http://[^ ]*"
```
Voila :)

**Comment by jeanleonino** [comment]
Is it possible to run from browser? Would be interesting as well and easy to spread.

**Comment by Theino** [comment]
> Let people attach their account to the fishnet servers their running, then give them a badge on their profile that shows how many analyses they've processed. That's motivation enough for me!

I think this would bring in a lot of people.

**Comment by FYPetro** [comment]
Disclosure: Very apparent conflict of interest here :-)

Assuming a contributor with lichess account has no malicious history, here is a concrete award plan for server donators, adapted from the posts above:

On the profile page, the admin shall enable one of the following scheme that awards server contributors with at least x million positions analyzed through fishnet:
  1. tiered trophy for linked accounts, at 1 million, 10 million and 100 million analyzed positions respectively; or
  2. integer badge for every million positions analyzed.

This statistic could also be coupled with the cloud analysis contributions.

**Comment by TheCodeArtist** [comment]
...or assign "Ranks" in increasing order of contribution from pawn, bishop, knight, rook, queen, to king!

The rank corresponding to recent(day/week/month) contribution would be more meaningful than an alltime rank as it would: 
- reward persistence over the short-term.
- encourage newcomers to contribute with a realistic chance of being on top.
   (as everyone starts from zero every day/week/month).

**Comment by mklemenz** [comment]
since this is a brainstorming section for an even larger fishnet: how about utilizing all the mobile devices out there?

arm cpus are not as powerful as x86/amd64. Plus battery drain is another huge issue with them. so here's my proposal:
A "mobile-fishnet" API endpoint would not give full games to analyze rather than just single positions. The server's responsibility would be to distribute them among the requesting devices and reconstruct the pgn afterwards. 
as for the mobile devices: the logic could be embedded within the lichess mobile application, since it's already wide spread. Opening the app would trigger (with some reasonable delay) a request for a new position or two to analyze, do the job and be done with it until the app is opened again. Since nobody likes battery hogs further restrictions could be put in place - eg: only when battery status is > 50% or loading. furthermore the user should be given a opt-in (or opt-out) possibility.

I don't know the numbers, if there're sufficient mobile clients out there, how often they open/close the app per day, etc. - but given the popularity of lichess I like to think this could work. Just think about all the unused computing power people are carrying in their pockets and handbags ;)

**Comment by SirMorfield** [comment]
One of the best ways to get people motivated is to make it a competition, make a page on lichess that displays a leader-board of  the best contributors. Something like this from _folding @ home_ (https://stats.foldingathome.org/teams)

**Comment by ornicar** [comment]
No, because then people would make crappy contributions (by sending weak evals) to get on the leaderboard.

**Comment by SirMorfield** [comment]
Then measure the 'score' in nodes crunched, so people can't cheat the system.

**Comment by ornicar** [comment]
You clearly underestimate what people will do to get on any leaderboard. I talk from some experience.

**Comment by SirMorfield** [comment]
If you say so.
So the difficulties of preventing cheating outweigh the benefits of a scoreboard idea, bummer.
Once again a project is ruined by cheaters.

**Comment by InnovativeInventor** [comment]
You could check for abuse by randomly re-doing the computation on a random selection of historical/current tasks for the people on the leaderboard. @niklasf If the results differ by too much, then you know they're cheating.

**Comment by niklasf** [comment]
That's a good idea. If we ever go to step 3 in https://github.com/niklasf/fishnet/milestones there will be no way around it.
